test/plots/barPlotTests.ts

```diff
@@ -1 +1,783 @@
+          barPlot.labelsEnabled(true);
+          let labels = barPlot.content().selectAll(".on-bar-label, .off-bar-label");
+          const centerValues = labels.select("text")[0].map((textNode) => {
+            const centerOfText = getCenterOfText(<SVGElement> textNode);
+            return valueScale.invert(isVertical ? centerOfText.y : centerOfText.x);
+          });
+          const wasOriginallyOnBar = labels[0].map((label) => d3.select(label).classed("on-bar-label"));
+
+          data.forEach((d, i) => {
+            const centerValue = centerValues[i];
+            valueScale.domain([centerValue - (DEFAULT_DOMAIN[1] - DEFAULT_DOMAIN[0]), centerValue]);
+            labels = barPlot.content().selectAll(".on-bar-label, .off-bar-label"); // re-select after rendering
+            if (wasOriginallyOnBar[i] && d.value < 0) {
+              assert.isTrue(d3.select(labels[0][i]).classed("off-bar-label"),
+                `cut-off on-bar label was switched to off-bar (index ${i})`);
+            } else {
+              const textNode = labels.select("text")[0][i];
+              assert.strictEqual(d3.select(textNode).style("visibility"), "hidden", `label for bar with index ${i} is hidden`);
+            }
+          });
+        });
+
+        // HACKHACK: This test is a bit hacky, but it seems to be testing for a bug fixed in
+        // https://github.com/palantir/plottable/pull/1240 . Leaving it until we find a better way to test for it.
+        it("removes labels instantly on dataset change", (done) => {
+          barPlot.labelsEnabled(true);
+          let texts = barPlot.content().selectAll("text");
+          assert.strictEqual(texts.size(), dataset.data().length, "one label drawn per datum");
+          const originalDrawLabels = (<any> barPlot)._drawLabels;
+          let called = false;
+          (<any> barPlot)._drawLabels = () => {
+            if (!called) {
+              originalDrawLabels.apply(barPlot);
+              texts = barPlot.content().selectAll("text");
+              assert.strictEqual(texts.size(), dataset.data().length, "texts were repopulated by drawLabels after the update");
+              called = true; // for some reason, in phantomJS, `done` was being called multiple times and this caused the test to fail.
+              done();
+            }
+          };
+          dataset.data(dataset.data());
+          texts = barPlot.content().selectAll("text");
+          assert.strictEqual(texts.size(), 0, "texts were immediately removed");
+        });
+
+        afterEach(function() {
+          if (this.currentTest.state === "passed") {
+            barPlot.destroy();
+            svg.remove();
+          }
+        });
+      });
+
+      describe(`retrieving Entities when ${orientation}`, () => {
+        const data = [
+          { base: -1, value: -1 },
+          { base: 0, value: 0 },
+          { base: 1, value: 1 }
+        ];
+        const DEFAULT_DOMAIN = [-2, 2];
```

Comment:
Yeah, larger and smaller than the min and max values. I could write code to generate it programmatically, but then a reader would have to verify that *that* code is correct too, which is harder than just looking at the values and seeing that they're bigger than the data values.